explainers/update-rendering.md

```diff
@@ -16,38 +22,62 @@ words, if the website intends to show an element whose contents are currently
 skipped, then skipping work may cause jank when the contents are ultimately
 presented.
 
-### Proposal
-The proposal is to add the following function, which lets the user-agent know
-that the element’s contents should be updated in preparation for display:
-
-Signature: `Promise Element.updateRendering(Priority?)`
+### Proposal: `renderPriority` element attribute
+When present, this attribute informs the User Agent that it should keep the
+element and its subtree updated whenever possible. The timing and frequency of
+updates are kept in the User Agent's control to allow flexibility in
+prioritizing this work.
 
-The function takes a Priority as an argument, with the following proposed values,
-following [the postTask API](https://wicg.github.io/scheduling-apis/#sec-task-priorities)
-example.
-
-* `user-blocking` is the highest priority, and is meant to be used for updates
+The values that the attribute takes inform the User Agent of developer intent:
+* `userBlocking` is the highest priority, and is meant to be used for updates
   that are blocking the user’s ability to interact with the page, such as
   rendering the core experience or responding to user input.
 
-* `user-visible` is the second highest priority, and is meant to be used for
+* `userVisible` is the second highest priority, and is meant to be used for
   updates that visible to the user but not necessarily blocking user actions,
   such as rendering secondary parts of the page. This is the default priority.
 
 * `background` is the lowest priority, and is meant to be used for updates that
   are not time-critical, such as background updates for speculative layout.
 
-The function returns a promise which resolves when the contents of the element
-are ready to be displayed to screen.  Note that repeated calls to
-updateRendering() returns new promises, with the ultimate priority of the
-request being the highest requested priority.
+* `auto` is the default priority, which should be equivalent to `userBlocking`
+  for the purposes of User Agent prioritization.
+
+Note that the attribute values follow [the postTask API](https://wicg.github.io/scheduling-apis/#sec-task-priorities)
+example.
+
+
+### Interaction with `content-visibility`
+Since the User Agent typically keeps rendering state of subtrees up-to-date,
+this feature would be a no-op in a majority of cases. For instance, having
+`renderPriority` on a visible, on screen, element would not have to do any
```

Comment:
I'm not sure about that.. I guess we haven't really discussed what userVisible means, but even if say it changes style and layout in separate frames, then that seems like it may be web breaking. I'm worried about making this enter the "render async" territory where things are visible but are not updated synchronously.

Or do you think that this is the opportunity to change this?

Reply:
> Or do you think that this is the opportunity to change this?

I think it's an opportunity to consider introducing async rendering across the board, not just in content-visibility. It would not be render breaking for existing content, because the `auto` behavior would be unchanged from today. We would also give UAs wiggle room to not implement it (e.g. we'd likely not do so in Chromium at first).

Reply:
I suggest that we move some of this discussion into an issue. I think that if a developer uses this in conjunction with `content-visibility`, and then at some point they remove `content-visibility`, they might not expect that although the element is now visible, it tends to be updated asynchronously:

```js
<div id=e style="content-visibility: hidden" renderPriority=background> ... </div>
...
e.style = "";
// e is now maybe asynchronous?
```

I think this text can change for sure, but coupling these two features in this explainer might complicated TAG/standards reviews. WDYT?

Reply:
ok sure let's put it in an issue. File one and link to it, with a reference to the behavior being an open discussion point?

Reply:
SGTM

Reply:
Done.